app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+    belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+    has_many :likes
+    has_many :comments    
+
+    def update_post_counter
+      author.update(posts_counter: author.posts.count)
+    end
+
+    def recent_post_comments
+      comments.order(created_at: :desc).limit(5)
+    end
+end
```

Comment:
- [ Optional ] Kindly I recommend you to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the` #update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private.

 ```
def recent_post_comments
      comments.order(created_at: :desc).limit(5)
    end

after_save :update_post_counter

  private

    def update_post_counter
      author.update(posts_counter: author.posts.count)
    end
```